tests/nexus_workflow_test.go

```diff
@@ -732,6 +733,225 @@ func (s *NexusWorkflowTestSuite) TestNexusOperationAsyncCompletion() {
 	s.False(seenCompletedEvent)
 }
 
+func (s *NexusWorkflowTestSuite) TestNexusOperationAsyncCompletionBeforeStart() {
+	ctx, cancel := context.WithCancel(testcore.NewContext())
```

Comment:
Why do you need the cancel function here?

Reply:
This test would be much more readable if you used the SDK. Please change.

Reply:
I added the cancel function because I wanted to make sure that the context passed to the background routine is canceled if the test exits early because of an assertion failure.

Reply:
I will rewrite the test to use the SDK in a followup.